modules/javafx.fxml/src/main/java/javafx/fxml/FXMLLoader.java

```diff
@@ -1699,9 +1766,14 @@ public void handle(Event event) {
             localBindings.put(scriptEngine.FILENAME, filename);
             // Execute the script
             try {
-                scriptEngine.eval(script, localBindings);
+                if (isCompiled) {
```

Comment:
I think there may be other places you need to set isCompiled (it isn't set in the first couple of methods where you compile scripts). Can you check this?

Reply:
isCompiled gets set explicitly to false at object creation time. It only will be changed to true, if the script was successfully compiled.

Reply:
I was (mistakenly) thinking of the `processStartElement` and `processEndElement` methods of the `ScriptElement` class, but that's a different class, and they use a local `compiledScript` variable. So this is fine.